### PR TITLE
[GPT-736] Split modules api

### PR DESCRIPTION
**Pull request by nickbalestra**
- [x] split-module API structure (template / template-compiler)
- [x] monorepo (lerna)
- [x] restructured tests for cleaner organization
- [x] invisible prettier workflow (no need for eslint for the moment, in the future prettier-eslint could be used if needed)

### CODE REVIEW

File: lerna.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "lerna": "2.0.0-rc.5",
```
Comment (matteofigus) [inline lerna.json]: 🎉

**Comment by nickbalestra** [comment]
Added moar tests (to make sure that the modules expose the correct APIs). @matteofigus @mattiaerre should be ready to be reviewed now

File: packages/oc-template-jade-compiler/compile.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    compiled = jade
+      .compileClient(options.template, {
+        compileDebug: false,
+        name: 't',
+        filename: options.viewPath
+      })
+      .toString()
+      .replace('function t(locals) {', 'function(locals){');
+  } catch (err) {
+    return callback(err);
```
Comment (mattiaerre) [inline packages/oc-template-jade-compiler/compile.js]: I sense that this path is not tested; have you tried to run the coverage?

`--coverage`

😅

Comment (nickbalestra) [inline packages/oc-template-jade-compiler/compile.js]: Should be covered now

**Comment by nickbalestra** [comment]
@mattiaerre , @matteofigus updated

File: packages/oc-template-jade-compiler/compile.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+const jade = require('jade-legacy');
+
+module.exports = (options, callback) => {
+  var compiled;
```
Comment (matteofigus) [inline packages/oc-template-jade-compiler/compile.js]: let

**Pull request merged**